### DISCUSSION TITLE
Not useful logs in vercel staging deploy

### DISCUSSION THREAD

**Discussion by aloukissas** [vercel/vercel#4118] [General] [answered]
Hi there

I'm getting a 500 for an SSR page that uses secrets via environment, although it works perfectly with `now dev` locally. The logs really don't have much to help me move forward. Any help appreciated:

`
11:14:36.007  
Error: Minified React error #152; visit https://reactjs.org/docs/error-decoder.html?invariant=152&args[]=Header for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
11:14:36.007  
    at Za (/zeit/376e481e/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:35:465)
11:14:36.007  
    at d (/zeit/376e481e/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:37:55)
11:14:36.007  
    at $a (/zeit/376e481e/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:39:16)
11:14:36.008  
    at a.b.render (/zeit/376e481e/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:44:476)
11:14:36.008  
    at a.b.read (/zeit/376e481e/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:44:18)
11:14:36.008  
    at renderToString (/zeit/376e481e/node_modules/react-dom/cjs/react-dom-server.node.production.min.js:54:364)
11:14:36.008  
    at render (/zeit/376e481e/node_modules/next/dist/next-server/server/render.js:83:16)
11:14:36.008  
    at Object.renderPage (/zeit/376e481e/node_modules/next/dist/next-server/server/render.js:406:16)
11:14:36.008  
    at Function.getInitialProps (/zeit/376e481e/.next/serverless/pages/index.js:1792:19)
11:14:36.008  
    at Object.loadGetInitialProps (/zeit/376e481e/node_modules/next/dist/next-server/lib/utils.js:59:29)
11:14:36.014  
Error occurred prerendering page "/". Read more: https://err.sh/next.js/prerender-error:
11:14:36.014  
Error: Failed to render serverless page
11:14:36.052  
> Build error occurred
11:14:36.056  
Error: Export encountered errors
11:14:36.056  
    at _default (/zeit/376e481e/node_modules/next/dist/export/index.js:19:1086)
11:14:36.056  
    at runMicrotasks (<anonymous>)
11:14:36.056  
    at processTicksAndRejections (internal/process/task_queues.js:94:5)
11:14:36.056  
    at async build (/zeit/376e481e/node_modules/next/dist/build/index.js:35:218)
`

**Comment by styfle** [comment, marked as answer]
Hi @aloukissas 

The error message has a couple links with more details.

https://reactjs.org/docs/error-decoder.html?invariant=152&args%5B%5D=Header

```
Header(...): Nothing was returned from render. This usually means a return statement is missing. Or, to render nothing, return null.
```

------


https://err.sh/next.js/prerender-error

#### Why This Error Occurred

While prerendering a page an error occurred. This can occur for many reasons from adding non-pages e.g. `components` to your `pages` folder or expecting props to be populated which are not.

#### Possible Ways to Fix It

- Make sure to move any non-pages out of the `pages` folder
- Check for any code that assumes a prop is available even when it might not be
- Check for any out of date modules that you might be relying on

**Reply by aloukissas** [reply]
I've looked at these links but nothing useful there, either. Page loads w/o issue in dev, which probably means that `now dev` isn't a 100% mirror of real deployment. I'll probably just switch to calling fetch within the component body. Sadly, debugging SSR pages has been the hardest part of using the platform (always cryptic error messages like these).

**Reply by styfle** [reply]
`now dev` is currently beta status but you could try `next dev` and see if that prints any errors.

Try removing `<Header>` and see if that fixes it.

**Reply by aloukissas** [reply]
`next dev` works equally well. Also there is no Header component in this page.

**Reply by styfle** [reply]
It sounds like you might have a dependency that's adding a `Header` because that doesn't come from Next.js

**Reply by aloukissas** [reply]
It's not in the tree. It's fine, it looks like there's work to be done to mirror local dev + deploy env and also help figure out errors in deploy. I'll switch to client-side fetch for now.

**Reply by aloukissas** [reply]
Even worse: now I still get a 500 w/o *any* logs :(

**Reply by aloukissas** [reply]
One more hint: I'm using now secrets, which I'm exposing in a `now.json` file. Do I also need a `next.config.js` file for this? The docs are a bit confusing.

**Reply by styfle** [reply]
Yes you will need to update `next.config.js`.

See this example https://github.com/zeit/next.js/tree/canary/examples/with-now-env

**Reply by styfle** [reply]
Also note that we recently announced the environment variable dashboard.

You can remove `now.json` and instead use the dashboard.

https://zeit.co/blog/environment-variables-ui

**Reply by aloukissas** [reply]
That's a bummer. Why remove the code-as-config option? It was great being able to config a project w/o going to the dashboard (and have it version-controlled).

**Reply by styfle** [reply]
> Why remove the code-as-config option?

This is not code, these are secrets. The values were never stored in `now.json`.

We improved the workflow from development to be `git pull` and `now env pull` and everyone on team gets the latest code and latest environment variables with secrets.
